### DISCUSSION TITLE
API: Get the current max context length for currently loaded model

### DISCUSSION THREAD

**Discussion by f-ale** [oobabooga/text-generation-webui#5317] [Q&A]
Hey everyone,
I am working on a client application which interfaces with text-generation-webui's new OpenAI API.
I'm currently generating new text using the completions endpoint. I am wondering if there is any way of retrieving the max context length for the currently loaded model via the API, so I can build my prompt accordingly and avoid going beyond that limit.

I haven't been able to find a method for that, am I missing something? What's the best way to avoid going beyond the model's context length?

Thank you!

**Comment by leighklotz** [comment]
It looks like a minimal local addition to text-generation-webui/extensions/openai/models.py would bring this to you:

```
def get_current_model_info():
    return {
        'model_name': shared.model_name,
        'lora_names': shared.lora_names,
        'loader': shared.args.loader,
+      'model_settings': get_model_metadata(shared.model_name)
    }
```

[Edit:] Also to make the http://localhost:5000/docs UI work right, add to text-generation-webui/extensions/openai/typing.py

```
from typing import Any
class ModelInfoResponse(BaseModel):
    model_name: str
    lora_names: List[str]
+   model_settings: Dict[str, Any]
```

**Comment by Technologicat** [comment]
Just bumped into this from a search.

I second the suggestion, would be very useful when coding custom apps that use `text-generation-webui` as an LLM backend.

The suggested solution looks nice. I suppose all we need now is a PR.
